tools/topology/CMakeLists.txt

```diff
@@ -157,6 +157,11 @@ foreach(tplg ${TPLGS})
 
 	add_custom_command(
 		OUTPUT ${output}.tplg
+# - In alsa-utils commit v1.2.2~15-gcbabe7a3f0cc, alsatplg (accidentally?)
+#   dropped the verbosity level; -v became a boolean.
+# - It unfortunately does not seem possible to base '-v' on ${VERBOSE}
+#   without adding a new layer of indirection / script between cmake and
+#   alsatplg
```

Comment:
Example: https://github.com/zephyrproject-rtos/zephyr/commit/4971d2a08452494d69

Reply:
This comment is not true. It is possible in many other ways. I'm not sure how you'd like to use it but if you want to base it on VERBOSE env variable, it can be done with parameter expressions or some subshells invoked inline. If you want to base it on alsa version then  you'd need to do some checks here in cmake.

If your problem is that you want to just be able to override this `-v 1` part, then we can add `VERBOSE_ALSA` env var switch for this. So it'd look like `COMMAND alsatplg \${VERBOSE_ALSA--v 1} -c ${output}.conf -o ${output}.tplg` and you can launch it with `VERBOSE_ALSA="" make ...`, to disable verbose.

Reply:
>  it can be done with parameter expressions 

To be portable, cmake COMMAND invokes `alsatplg` directly, it doesn't go through an OS dependent shell, not unless you explicitly ask for a non-portable shell

> So it'd look like COMMAND alsatplg \${VERBOSE_ALSA--v 1} 

First thing I naively tried.

> some subshells invoked inline

= adding a new (subshell) layer of indirection. I can rephrase to include that.



> you'd need to do some checks here in cmake.

VERBOSE works _after_ cmake has run, that's the beauty of it. So yes, you can probably decide this at cmake time (I can give it a try) but not at make/ninja time because by then cmake has run, it's too late.

> This comment is not true. 

I suggest  a closer look at the level of cmake wizardry in the Zephyr project before assuming 
https://github.com/zephyrproject-rtos/zephyr/commit/4971d2a08452494d69 didn't know what they were doing :-)

Reply:
> > some subshells invoked inline
> 
> = adding a new (subshell) layer of indirection. I can rephrase to include that.

Well, yea, you can think of it as level of indirection ;)

However, you still can do this with params expressions, they just need to be properly escaped - You can disable VERBATIM if it helps. We support windows only for stuff that is really done there like building FW, I guess no1 is going to run alsa tools on windows and I was adding VERBATIM usually cos of cross-platform support.
I didn't try the sample that I wrote (`\${VERBOSE_ALSA--v 1}`), it was just hint . You may need to escape `$` as `$$` instead of `\$`.